src/js/components/field.js

```diff
@@ -353,11 +353,32 @@ export default class Field {
               attrs: typeAttrs(key, val, 'string'),
               action: {
                 change: evt => {
+                  // This is a workaround for options attributes fields.
+                  // Root cause of the problem is that fMap is calculated when the field is created
+                  // but it is not updated when collection of options is changed (e.g removal of the option).
+                  // As the result fMap contains outdated option index pointing to different element or to undefined.
+                  // The workaround replaces initial option index using current position od <li> element in DOM.
+                  // NOTE: The workaround will most likely stop working or produce exceptions when DOM structure changes.
+                  if (fMap.startsWith('options')) {
```

Comment:
I'm not sure this is the right place for this change but will have a better idea after testing. Maybe we could move it to [common/data.js#L100](https://github.com/Draggable/formeo/blob/master/src/js/common/data.js#L100)? This would enable us to save option order without reading `fMap` and make it part of the save process so other actions will get the benefit.

We might also need to add `onRemove` to [components/panels.js#L117](https://github.com/Draggable/formeo/blob/master/src/js/components/panels.js#L117) to trigger the `saveOptionsOrder`

Thanks for the PR and flagging this problem. I love seeing these contributions :)

Reply:
I agree it's not the best solution for the problem. I resolved the problem as a workaround because it was urgent to fix. 

I cannot promise anything, but I'll try to come up with something better in the future. Cheers!